### PR TITLE
Use librustzcash proto-generated compactformats instead of our own proto-generated code

### PR DESCRIPTION
**Pull request by AloeareV**

### CODE REVIEW

**Review by zancas — DISMISSED** [review]
Fantastic!

**Review by zancas — COMMENTED** [review]
This is great!   Benchmarking and testing locally.

**Comment by zancas** [comment]
I see this very interesting test failure, but it's against a rebased version:

```
        FAIL [  24.521s] darkside-tests::advanced_reorg_tests reorg_changes_outgoing_tx_index

--- STDOUT:              darkside-tests::advanced_reorg_tests reorg_changes_outgoing_tx_index ---

running 1 test
dataset path: /home/nattyb/src/zingolabs/zingolibs/librustzcash_compactformats/darkside-tests/tests/data/wallets/advanced_reorg_tests/expires_incoming_tx/before_reorg.txt
SENT TX ID: "408303d91c28c82167125221b00978fe0dd983cb4b97b3b0383dd292e6767b67"
[ValueTransfer { block_height: BlockHeight(203), datetime: 1, kind: Received { pool: Orchard, amount: 100000000 }, memos: [], price: None, txid: TxId("8698b158c0746e868615611ad9ca106465b51032a7c5c45a1963f5a193f5cce1"), unconfirmed: false }, ValueTransfer { block_height: BlockHeight(205), datetime: 1701841195, kind: Sent { amount: 100000, to_address: ZcashAddress { net: Regtest, kind: Unified(Address([P2pkh([194, 233, 133, 149, 15, 205, 125, 169, 93, 133, 49, 242, 127, 97, 240, 45, 186, 96, 1, 45]), Sapling([85, 56, 108, 149, 244, 208, 77, 10, 114, 22, 236, 103, 220, 254, 110, 55, 215, 13, 12, 101, 127, 190, 95, 98, 191, 226, 249, 105, 161, 0, 248, 79, 70, 73, 206, 64, 206, 2, 229, 226, 181, 177, 183]), Orchard([202, 101, 1, 126, 115, 154, 198, 112, 65, 23, 144, 213, 80, 49, 5, 41, 212, 132, 38, 87, 245, 237, 89, 140, 206, 126, 190, 166, 107, 199, 153, 7, 235, 13, 152, 24, 56, 245, 151, 188, 214, 97, 147])])) } }, memos: [], price: None, txid: TxId("61707f2560d0fa144db012d7d0bdc8d0c2d26269e5695eb2fb2e5c3b13463a16"), unconfirmed: true }, ValueTransfer { block_height: BlockHeight(205), datetime: 1701841195, kind: Fee { amount: 10000 }, memos: [], price: None, txid: TxId("61707f2560d0fa144db012d7d0bdc8d0c2d26269e5695eb2fb2e5c3b13463a16"), unconfirmed: true }]
{ success: true, latest_block: 211, total_blocks_synced: 6}
test reorg_changes_outgoing_tx_index ... FAILED

failures:

failures:
    reorg_changes_outgoing_tx_index

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 7 filtered out; finished in 24.50s


--- STDERR:              darkside-tests::advanced_reorg_tests reorg_changes_outgoing_tx_index ---
[darkside-tests/tests/advanced_reorg_tests.rs:685] &uri = http://127.0.0.1:22637/
thread 'reorg_changes_outgoing_tx_index' panicked at darkside-tests/tests/advanced_reorg_tests.rs:1069:5:
assertion `left == right` failed
  left: PoolBalances { sapling_balance: Some(0), verified_sapling_balance: Some(0), spendable_sapling_balance: Some(0), unverified_sapling_balance: Some(0), orchard_balance: Some(100000000), verified_orchard_balance: Some(100000000), unverified_orchard_balance: Some(0), spendable_orchard_balance: Some(100000000), transparent_balance: Some(0) }
 right: PoolBalances { sapling_balance: Some(0), verified_sapling_balance: Some(0), spendable_sapling_balance: Some(0), unverified_sapling_balance: Some(0), orchard_balance: Some(99890000), verified_orchard_balance: Some(99890000), unverified_orchard_balance: Some(0), spendable_orchard_balance: Some(99890000), transparent_balance: Some(0) }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by zancas** [comment]
mob-prerelease-1 3 0-519-g504f6be4_sync_duration_annotation

**Review by zancas — DISMISSED** [review]
W000000T!!!

**Comment by AloeareV** [comment]
Not sure why the CI build is failing, it builds locally.

**Comment by AloeareV** [comment]
@Oscar-Pepper could you shed some light on this? The tests build and pass locally, so I'm assuming some part of the CI script is aware of the location of the proto files, or the proto files aren't accounted for for the sake of a cashe hit and so we're getting a false hit and not rebuilding...or something? I'm not really sure what's going on here, just that a move of the proto files seems to have caused a build failure on CI that it doesn't cause locally. 

One option is to just...put the proto files back, and have them be in zingolib/zingolib despite them only being in tests..that would be the simplest solution, probably.

**Comment by Oscar-Pepper** [comment]
it looks like the test runner needs protofbuf-compiler installed but i dont understand whats changed to have not needed it before. i will send a PR with the fix

**Comment by AloeareV** [comment]
> it looks like the test runner needs protofbuf-compiler installed but i dont understand whats changed to have not needed it before. i will send a PR with the fix

Yeah I'm befuddled as to why it's needed now and not before...grabbefd your fix, rerunning CI.

File: zingolib/src/wallet/traits.rs [hunk 1]
```diff
@@ -257,18 +265,25 @@ impl Recipient for zcash_primitives::sapling::PaymentAddress {
     }
 }
 
-pub trait CompactOutput<D: DomainWalletExt>:
-    Sized + ShieldedOutput<D, COMPACT_NOTE_SIZE> + Clone
+fn slice_to_array<const N: usize>(slice: &[u8]) -> &[u8; N] {
+    <&[u8; N]>::try_from(slice).unwrap_or(&[0; N])
+    //todo: This default feels dangerous. Find better solution
```
Comment (Oscar-Pepper) [inline zingolib/src/wallet/traits.rs:270]: dangerous enough to not merge into dev yet??

Comment (AloeareV) [reply zingolib/src/wallet/traits.rs:270]: This is already in dev, this is a move https://github.com/zingolabs/zingolib/pull/743/files#diff-85a6f01ada5ff7e4b57dcd0d86ded00dd41c6f7f2e05b0367fc2e2c84d3c3cd9L111-L114 and not new logic.

**Review by Oscar-Pepper — APPROVED** [review]
Wonderful!

**Pull request merged**